clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1697,18 +1697,73 @@ private PushJobDetailsStatusTuple getPushJobDetailsStatusTuple(int status) {
     return tuple;
   }
 
-  private void sendPushJobDetailsToController() {
+  /**
+   * Tracks the last reported status to prevent duplicate terminal failure updates to the controller.
+   * Duplicate terminal statuses (e.g., ERROR, KILLED) may be misinterpreted in controller-side metrics,
+   * potentially skewing push job SLO calculations and causing confusion.
+   */
+  private volatile PushJobDetailsStatus lastReportedStatus = PushJobDetailsStatus.UNKNOWN;
+  private final Object lastReportedStatusLock = new Object();
+
+  private static PushJobDetailsStatus getCurrentOverallStatus(PushJobDetails pushJobDetails) {
+    if (pushJobDetails != null && !pushJobDetails.overallStatus.isEmpty()) {
+      return PushJobDetailsStatus
+          .valueOf(pushJobDetails.overallStatus.get(pushJobDetails.overallStatus.size() - 1).status);
+    }
+    return PushJobDetailsStatus.UNKNOWN;
+  }
+
+  /**
+   * Determines whether the push job status update should be skipped.
+   *
+   * <p>This method checks for several conditions that would make a status update unnecessary or
+   * potentially harmful:
+   * <ul>
+   *   <li>If status uploads are disabled via config, the update is skipped and a warning is logged
+   *       (only once).</li>
+   *   <li>If the push job details are not initialized, the update is skipped.</li>
+   *   <li>If the last reported status was a terminal failure (ERROR or KILLED) and the current status
+   *       is also a terminal failure, the update is skipped to avoid redundant controller updates,
+   *       which may negatively affect downstream metrics.</li>
+   * </ul>
+   *
+   * @return {@code true} if the status update should be skipped, {@code false} otherwise.
+   */
+  private boolean shouldSkipPushJobStatusUpdate() {
+    // Check if status upload is enabled
     if (!pushJobSetting.enablePushJobStatusUpload) {
```

Comment:
`enablePushJobStatusUpload` is enabled in all prod plugins, remove the check and delete the config?

Reply:
Done!